### DISCUSSION TITLE
wallet connect v1

### DISCUSSION THREAD

**Discussion by ryuk8000** [WalletConnect/walletconnect-monorepo#2412] [Developer Support] [answered]
i was using cdn of @walletconnect/web3-provider. will this also get deprecated after 28 June 2023.

I have also compiled this @walletconnect/web3-provider   using webpack and using the package in html file. How will package getting deprecated affect me if i already compiled the working stage of package and using it

**Comment by nikitaNotFound** [comment]
Hi! I believe you have to use etherium-provider or universal-provider.

https://docs.walletconnect.com/2.0/web/providers/ethereum
https://docs.walletconnect.com/2.0/web/providers/universal

Here is some code snippet. Hope I got your question correctly.
```
const provider = await UniversalProvider.init({
   logger: "debug",
   projectId: projectId,
   metadata: {
       name: '...',
       description: 'Sample app to test WalletConnect.',
       url: '...',
       icons: ['...']
   },
   client: signClient
});
await provider.connect({
   namespaces
});

const web3Provider = new ethers.BrowserProvider(provider);
```

**Comment by finessevanes** [comment, marked as answer]
Anything using v1 will not be working after June 28. Please be sure to have migrated by then. Here are the [migration docs](https://docs.walletconnect.com/2.0/advanced/migration-from-v1.x/dapps#web3-provider)
